sktime/forecasting/statsmodels_arima.py

```diff
@@ -0,0 +1,316 @@
+        approximation of the Hessian matrix, 'robust' for an approximate
+        (quasi-maximum likelihood) covariance matrix that may be valid even in the
+         presence of some misspecifications. Intermediate calculations use the 'oim'
+         method. 'robust_approx' is the same as 'robust' except that the intermediate
+         calculations use the 'approx' method. 'none' for no covariance matrix
+         calculation. Default is 'opg' unless memory conservation is used to avoid
+         computing the loglikelihood values for each observation,
+         in which case the default is 'oim'.
+
+    cov_kwds : dict or None, optional
+        A dictionary of arguments affecting covariance matrix computation.
+
+    return_params : bool, optional
+        Whether to return only the array of maximizing parameters. Default is False.
+
+    low_memory : bool, optional If set to True, techniques are applied to
+        substantially reduce memory usage. If used, some features of the results object
+        will not be available (including smoothed results and in-sample prediction),
+        although out-of-sample forecasting is possible. Default is False.
+
+    See Also
+    --------
+    ARIMA
+    AutoARIMA
+    StatsForecastAutoARIMA
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.forecasting.statsmodels_arima import StatsModelsARIMA
+    >>> y = load_airline()
+    >>> forecaster = StatsModelsARIMA(order=(0, 0, 12))  # doctest: +SKIP
+    >>> forecaster.fit(y)  # doctest: +SKIP
+    >>> y_pred = forecaster.predict(fh=[1,2,3])  # doctest: +SKIP
+    """
+
+    _tags = {
+        # packaging info
+        # --------------
+        "authors": ["arnaujc91"],
+        "maintainers": ["arnaujc91"],
+        "ignores-exogeneous-X": False,
+        "capability:pred_int": True,
+        "capability:pred_int:insample": True,
+        "python_dependencies": ["statsmodels"],
+    }
+
+    def __init__(
+        self,
+        order: Tuple[int, int, int] = (0, 0, 0),
+        seasonal_order: Tuple[int, int, int, int] = (0, 0, 0, 0),
+        trend: Optional[Union[str, Iterable]] = None,
+        enforce_stationarity: bool = True,
+        enforce_invertibility: bool = True,
+        concentrate_scale: bool = False,
+        trend_offset: int = 1,
+        dates: Optional[np.ndarray] = None,
+        freq: Optional[str] = None,
+        missing: Optional[str] = None,
+        validate_specification: bool = True,
+        start_params: Optional[np.ndarray] = None,
+        transformed: bool = True,
+        includes_fixed: bool = False,
+        method: Optional[str] = None,
+        method_kwargs: Optional[dict] = None,
+        gls: Optional[bool] = False,
+        gls_kwargs: Optional[dict] = None,
+        cov_type: Optional[str] = "opg",
+        cov_kwds: Optional[dict] = None,
+        return_params: Optional[bool] = False,
+        low_memory: Optional[bool] = False,
```

Comment:
Same as above. `Optional[bool]` sort of suggests `None` is a valid value to pass I think.

Reply:
I do not think it suggest that `None` is valid, i understand where you get this conclusion though. Optional just means that there is a default value, as the variable will be checked at some point in the code if it is used. I get that you assume that the default is `None` as it happens often in the code, but i disagree that `Optional[bool]` suggests that u can pass `None` as a parameter.

Reply:
> Optional just means that there is a default value

@arnaujc91 Sorry, I disagree. My understanding comes from following ([reference](https://docs.python.org/3/library/typing.html#typing.Optional)):

> typing.Optional
> 
>     Optional[X] is equivalent to X | None (or Union[X, None]).
> 
>     Note that this is not the same concept as an optional argument, which is one that has a default. An optional argument with a default does not require the Optional qualifier on its type annotation just because it is optional. For example:
> 
>     def foo(arg: int = 0) -> None:
>         ...
> 
>     On the other hand, if an explicit value of None is allowed, the use of Optional is appropriate, whether the argument is optional or not. For example:
> 
>     def foo(arg: Optional[int] = None) -> None:
>         ...
> 
>     Changed in version 3.10: Optional can now be written as X | None. See [union type expressions](https://docs.python.org/3/library/stdtypes.html#types-union).

Please let me know if you think I am misinterpreting this.

Reply:
Hey @yarnabrina , no it is actually clear that i was wrong:

```
Optional[X] is equivalent to X | None (or Union[X, None]).
```

 I never checked that documentation, so thank you for letting me know! I will change this to remove `Optional`.